src/commands/admin/role.ts

```diff
@@ -0,0 +1,177 @@
+import { TextChannel, Message, ApplicationCommandOptionData, CommandInteraction, MessageEmbed, MessageActionRow,
+	MessageSelectOptionData, MessageSelectMenu, Client, Interaction, GuildMember } from 'discord.js';
+import { Command } from '@lib/types/Command';
+import { BOT } from '@root/config';
+
+const modes = ['Add', 'Remove'];
+
+export default class extends Command {
+
+	description = `Add/remove a role to a message's role dropdown`;
+	usage = '<messageID> | <channel> | <mode> | <role>';
+
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'messageid',
+			description: 'The message to add the role dropdown to',
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'channel',
+			description: 'The channel your target message was sent in',
+			type: 'CHANNEL',
+			required: true
+		},
+		{
+			name: 'mode',
+			description: `Add or remove a role to this message's role dropdown?`,
+			type: 'STRING',
+			required: true,
+			choices: modes.map((arg) => ({
+				name: arg,
+				value: arg
+			}))
+		},
+		{
+			name: 'role',
+			description: 'The role to add/remove to the dropdown',
+			type: 'ROLE',
+			required: true
+		}
+	]
+
+	run(_msg: Message): Promise<void> { return; }
+
+	async tempRun(interaction: CommandInteraction): Promise<void> {
+		const channel = interaction.options.getChannel('channel') as TextChannel;
+		if (!channel || channel.type !== 'GUILD_TEXT') {
+			const responseEmbed = new MessageEmbed()
+				.setColor('#ff0000')
+				.setTitle('Argument error')
+				.setDescription(`You must tag a text channel.`);
+			return interaction.reply({ embeds: [responseEmbed], ephemeral: true });
```

Comment:
You have this duplicated many times. Consider making it a function to condense the code a bit.